### PR TITLE
Feature/bitcoin cash

### PR DESCRIPTION
**Pull request by riaanjvos**

### CODE REVIEW

File: src/main/java/info/blockchain/wallet/api/data/EthereumOptions.java [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+@JsonIgnoreProperties(ignoreUnknown = true)
+@JsonAutoDetect(fieldVisibility = Visibility.NONE,
+    getterVisibility = Visibility.NONE,
+    setterVisibility = Visibility.NONE,
+    creatorVisibility = Visibility.NONE,
+    isGetterVisibility = Visibility.NONE)
+public class EthereumOptions {
+
+    @JsonProperty("lastTxFuse")
+    private int lastTxFuse;
```
Comment (ditn) [inline src/main/java/info/blockchain/wallet/api/data/EthereumOptions.java:21]: Bit weird that we parse as `int` and then return as `long` but I don't suppose it matters

Comment (riaanjvos) [reply src/main/java/info/blockchain/wallet/api/data/EthereumOptions.java:21]: It is a bit weird. Doesn't matter for now, but will update it for next release

File: src/main/java/info/blockchain/wallet/util/PrivateKeyFactory.java [hunk 2]
```diff
@@ -108,20 +109,22 @@ private ECKey decodeMiniKey(String mini) throws Exception {
...
+    // TODO: 04/01/2018 Pass params in. What about other coin params?
     private ECKey determineKey(String hash) throws Exception {
 
         ECKey uncompressedKey = decodeHexPK(hash, false);
         ECKey compressedKey = decodeHexPK(hash, true);
 
         try {
-            String uncompressedAddress = uncompressedKey.toAddress(PersistentUrls.getInstance().getCurrentNetworkParams()).toString();
-            String compressedAddress = compressedKey.toAddress(PersistentUrls.getInstance().getCurrentNetworkParams()).toString();
+            String uncompressedAddress = uncompressedKey.toAddress(PersistentUrls.getInstance().getBitcoinParams()).toString();
```
Comment (ditn) [inline src/main/java/info/blockchain/wallet/util/PrivateKeyFactory.java:119]: This is fine for now but probably needs updating fairly soon

File: src/main/java/info/blockchain/wallet/util/Tools.java [hunk 3]
```diff
@@ -101,10 +97,10 @@ public static ECKey getECKeyFromKeyAndAddress(@Nonnull String decryptedKey, @Non
             keyUnCompressed = ECKey.fromPrivate(priv2, false);
         }
 
-        if (keyCompressed.toAddress(PersistentUrls.getInstance().getCurrentNetworkParams())
+        if (keyCompressed.toAddress(PersistentUrls.getInstance().getBitcoinParams())
```
Comment (ditn) [inline src/main/java/info/blockchain/wallet/util/Tools.java:100]: Same here

**Review by ditn — APPROVED** [review]

**Pull request merged**